dp-hello-world-controller/features/steps/example_component.go

```diff
@@ -0,0 +1,116 @@
+package steps
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/ONSdigital/dp-api-clients-go/v2/health"
+	componenttest "github.com/ONSdigital/dp-component-test"
+	"github.com/ONSdigital/dp-healthcheck/healthcheck"
+	"github.com/ONSdigital/dp-hello-world-controller/assets"
+	"github.com/ONSdigital/dp-hello-world-controller/config"
+	"github.com/ONSdigital/dp-hello-world-controller/service"
+	"github.com/ONSdigital/dp-hello-world-controller/service/mocks"
+	dphttp "github.com/ONSdigital/dp-net/http"
+	render "github.com/ONSdigital/dp-renderer"
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+type Component struct {
+	componenttest.ErrorFeature
+	Config         *config.Config
+	killChannel    chan os.Signal
+	HTTPServer     *http.Server
+	ServiceRunning bool
+	svc            *service.Service
+	svcErrors      chan error
+	StartTime      time.Time
+	uiFeature      *componenttest.UIFeature
+}
+
+// NewComponent creates and returns a new component instance.
+func NewComponent() (*Component, error) {
+	c := &Component{
+		HTTPServer:     &http.Server{},
+		svcErrors:      make(chan error, 1),
+		ServiceRunning: false,
+	}
+
+	ctx := context.Background()
+
+	cfg, err := config.Get()
+	if err != nil {
+		return nil, err
+	}
+
+	c.Config = cfg
+
+	initMock := &mocks.InitialiserMock{
+		DoGetHTTPServerFunc:   c.DoGetHTTPServer,
+		DoGetHealthCheckFunc:  c.DoGetHealthCheck,
+		DoGetHealthClientFunc: c.DoGetHealthClient,
+	}
+
+	serviceList := service.NewServiceList(initMock)
+
+	renderer := render.NewWithDefaultClient(assets.Asset, assets.AssetNames, cfg.PatternLibraryAssetsPath, cfg.SiteDomain)
+
+	c.svc = service.New()
+	if err := c.svc.Init(ctx, c.Config, serviceList, renderer); err != nil {
+		log.Error(ctx, "failed to initialise service", err)
+		return nil, err
+	}
+
+	c.StartTime = time.Now()
+	c.svc.Run(ctx, c.svcErrors)
+	c.ServiceRunning = true
+
+	c.uiFeature = componenttest.NewUIFeature("http://" + c.Config.SiteDomain + c.Config.BindAddr)
+
+	return c, nil
+}
+
+// Closes closes down component resources. Called once all tests have been run.
+func (c *Component) Close() error {
```

Comment:
Just a question, will this function ever return an error?

Reply:
So `c.uiFeature.Close()` could potentially return a error and any other resource that's added by the user to `Component` after this project is generated so I thought it was better to set this to return an error in the generator.  Rather than no return value and then this being overlooked down the line in generated projects.